src/app/3d/qgs3dmapcanvasdockwidget.cpp

```diff
@@ -291,6 +297,34 @@ void Qgs3DMapCanvasDockWidget::configure()
   mBtnMapThemes->setDisabled( map->terrainGenerator()->type() == QgsTerrainGenerator::Mesh );
 }
 
+void Qgs3DMapCanvasDockWidget::exportScene()
+{
+
+  QDialog dlg;
+  dlg.setWindowTitle( tr( "Export 3D Scene" ) );
+  dlg.setObjectName( QStringLiteral( "3DSceneExportDialog" ) );
+  dlg.setMinimumSize( 380, 460 );
+//  QgsGui::instance()->enableAutoGeometryRestore( &dlg );
+
+  Qgs3DMapExportSettings *exportSettings = new Qgs3DMapExportSettings( &dlg );
```

Comment:
Is this deleted anywhere? Maybe better to just use stack-based local variable: `Qgs3DMapExportSettings exportSettings( &dlg );`

Reply:
I made the dialog object a parent of that class, so it is deleted when the dlg object is deallocated. I guess I was over engineering here.